src/main/java/io/vertx/ext/mongo/impl/MongoIterableStream.java

```diff
@@ -92,7 +94,12 @@ public MongoIterableStream resume() {
         return this;
       }
     }
-    queue.resume();
+    if (queue.isEmpty() && closeShouldBeCalled) {
```

Comment:
`queue.resume()` is called in both branches, this can be simplified.

Reply:
good notice :)